backend/app.py

```diff
@@ -133,7 +133,7 @@ def delete_hierarchy(hierarchy_id):
 
 @app.route("/node/<node_id>", methods=['DELETE'])
```

Comment:
I would expect this to have the same route as creating a node: "/hierarchy/<hierarchy_id>/node/<node_id>". If the node exists but doesn't match the hierarchy_id then you return a 404.